app/lib/client_sorter.rb

```diff
@@ -36,8 +37,10 @@ def filtered_clients
               else
                 @clients.after_consent
               end
-    # Force an inner join to `intakes` to exclude clients from previous years
-    clients = clients.joins(:intake)
+    # Filter on product_year to only show clients who used this-year's product
+    if @use_product_year
+      clients = clients.where(intake: Intake.where(type: "Intake::CtcIntake", product_year: MultiTenantService.new(:ctc).current_product_year).or(Intake.where(type: "Intake::GyrIntake", product_year: MultiTenantService.new(:gyr).current_product_year)))
```

Comment:
Some initial testing:

Before:

```
SELECT "clients"."id", "clients"."attention_needed_since", "clients"."completion_survey_sent_at", "clients"."consented_to_service_at", "clients"."created_at", "clients"."ctc_experience_survey_sent_at", "clients"."ctc_experience_survey_variant", "clients"."current_sign_in_at", "clients"."current_sign_in_ip", "clients"."experience_survey", "clients"."failed_attempts", "clients"."filterable_number_of_required_documents", "clients"."filterable_number_of_required_documents_uploaded", "clients"."filterable_percentage_of_required_documents_uploaded", "clients"."filterable_tax_return_properties", "clients"."first_unanswered_incoming_interaction_at", "clients"."flagged_at", "clients"."identity_verification_denied_at", "clients"."identity_verified_at", "clients"."in_progress_survey_sent_at", "clients"."last_13614c_update_at", "clients"."last_incoming_interaction_at", "clients"."last_internal_or_outgoing_interaction_at", "clients"."last_outgoing_communication_at", "clients"."last_seen_at", "clients"."last_sign_in_at", "clients"."last_sign_in_ip", "clients"."locked_at", "clients"."login_requested_at", "clients"."login_token", "clients"."message_tracker", "clients"."needs_to_flush_filterable_properties_set_at", "clients"."previous_sessions_active_seconds", "clients"."restricted_at", "clients"."routing_method", "clients"."sign_in_count", "clients"."still_needs_help", "clients"."triggered_still_needs_help_at", "clients"."updated_at", "clients"."vita_partner_id" FROM "clients" WHERE "clients"."consented_to_service_at" IS NOT NULL ORDER BY "clients"."last_outgoing_communication_at" ASC LIMIT 50 OFFSET 0;
Time: 33.278 ms
```

After

```
SELECT "clients"."id", "clients"."attention_needed_since", "clients"."completion_survey_sent_at", "clients"."consented_to_service_at", "clients"."created_at", "clients"."ctc_experience_survey_sent_at", "clients"."ctc_experience_survey_variant", "clients"."current_sign_in_at", "clients"."current_sign_in_ip", "clients"."experience_survey", "clients"."failed_attempts", "clients"."filterable_number_of_required_documents", "clients"."filterable_number_of_required_documents_uploaded", "clients"."filterable_percentage_of_required_documents_uploaded", "clients"."filterable_tax_return_properties", "clients"."first_unanswered_incoming_interaction_at", "clients"."flagged_at", "clients"."identity_verification_denied_at", "clients"."identity_verified_at", "clients"."in_progress_survey_sent_at", "clients"."last_13614c_update_at", "clients"."last_incoming_interaction_at", "clients"."last_internal_or_outgoing_interaction_at", "clients"."last_outgoing_communication_at", "clients"."last_seen_at", "clients"."last_sign_in_at", "clients"."last_sign_in_ip", "clients"."locked_at", "clients"."login_requested_at", "clients"."login_token", "clients"."message_tracker", "clients"."needs_to_flush_filterable_properties_set_at", "clients"."previous_sessions_active_seconds", "clients"."restricted_at", "clients"."routing_method", "clients"."sign_in_count", "clients"."still_needs_help", "clients"."triggered_still_needs_help_at", "clients"."updated_at", "clients"."vita_partner_id" FROM "clients" WHERE "clients"."consented_to_service_at" IS NOT NULL AND "clients"."id" IN (SELECT "intakes"."client_id" FROM "intakes" WHERE "intakes"."product_year" = 2022 AND ("intakes"."type" = 'Intake::CtcIntake' OR "intakes"."type" = 'Intake::GyrIntake')) ORDER BY "clients"."last_outgoing_communication_at" ASC LIMIT 50 OFFSET 0;
```
still executing, been > 10 sec

Makes you wonder if a JOIN would help though

Also makes you wonder if an `intakes` index would help